rdt-app/src/main/java/io/ona/rdt_app/activity/PatientRegisterActivity.java

```diff
@@ -137,7 +140,7 @@ protected void onActivityResult(int requestCode, int resultCode, @Nullable Inten
     @Override
     public void onFormSaved() {
         try {
-            Patient rdtPatient = patientRegisterFragmentPresenter.getRDTPatient();
+            Patient rdtPatient = patientRegisterFragmentPresenter.getRDTPatient(jsonForm);
```

Comment:
Synchronized keyword should be on the `getRDTPatient` method so that no thread can alter global variables while it is processing them?

Reply:
It can also remain on the `saveForm` block